### ISSUE TITLE
If group which has vhosts offered is removed, the vhosts aren't removed which causes error messages

### ISSUE THREAD

**Issue by Mikaela** [atheme/atheme#428]
I created group `!test` and offered it two vhosts: `test/$account` and `tester/$account` and then removed the group.

I still see them in `/hs offerlist`

```
2015-01-10 15:21:40+0200 -- HostServ: vhost:test/$account, group:(null) creator:Mikaela (Jan 10 10:40:43 2015)
2015-01-10 15:21:40+0200 -- HostServ: vhost:tester/$account, group:(null) creator:Mikaela (Jan 10 10:40:45 2015)
```

and OperServ says

````
2015-01-10 15:21:40+0200 <@OperServ> (offer.c:249 myuser_is_in_group): warning: assertion '(mg = ((mt != ((void *)0) && ((myentity_t *)(mt))->type == ENT_GROUP) ? (mygroup_t *)(mt) : ((void *)0))) != ((void *)0)' failed.
2015-01-10 15:21:40+0200 <@OperServ> (offer.c:249 myuser_is_in_group): warning: assertion '(mg = ((mt != ((void *)0) && ((myentity_t *)(mt))->type == ENT_GROUP) ? (mygroup_t *)(mt) : ((void *)0))) != ((void *)0)' failed.
```

VERSION atheme 7.2.3. services.queertech.irc 470708d45a230d0564854bcd57e9bb05be1b4161 :cFljR [InspIRCd] [enc:posix] Build Date: Jan  7 2015

**Comment by Renegade334** [comment]
This actually manifested itself on our network as a database corruption, where the NULL group pointer was saved in the `HO` database line as `*` causing Atheme to exit on load.

Committed to candidate branch, comments welcome.

**Comment by kaniini** [comment]
It seems fine, feel free to pull-request it.

**Comment by kaniini** [comment]
Fixed by #431.
